packages/esm-patient-registration-app/src/patient-registration/field/person-attributes/location-person-attribute-field.resource.tsx

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useMemo } from 'react';
+import { type FetchResponse, fhirBaseUrl, openmrsFetch, useDebounce } from '@openmrs/esm-framework';
+import { type LocationEntry, type LocationResponse } from '@openmrs/esm-service-queues-app/src/types';
+import useSWR from 'swr';
+interface IUseLocations {
+  locations: Array<LocationEntry>;
+  isLoading: boolean;
+  loadingNewData: boolean;
+  fetchLocationByUuid: (uuid: string) => void;
+}
+
+export function useLocations(locationTag: string | null, count: number = 0, searchQuery: string = ''): IUseLocations {
+  const debouncedSearchQuery = useDebounce(searchQuery);
+
+  const fetchLocationByUuid = useCallback((uuid: string) => {
+    const { data } = useSWR<FetchResponse<LocationResponse>, Error>(`${fhirBaseUrl}/Location/${uuid}`, openmrsFetch);
+  }, []);
```

Comment:
Don't use `useSWR()`. It's a violation of the [rules of hooks](https://react.dev/reference/rules/rules-of-hooks).